### PR TITLE
RHTAP-1494: Verify tenants configurations

### PR DESCRIPTION
**Pull request by gbenhaim**
Add tests for verifying that each tenant name ends with "-tenant" and for ensuring all the resources belongs to a tenant directory, are going to be deployed to a namespace with the same name.

In addition, add Python tooling (the tests were implemented in Python), and a Github action for running the tests in the CI.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: tests
+
+on:
+  pull_request:
+    branches: [ main ]
+jobs:
+  test:
+    permissions: write-all
```
Comment (ernesgonzalez33) [inline .github/workflows/test.yaml:8]: Do we really need `write all` permissions for this job?

File: tests/test_manifests.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        """The definition of the resource"""
+
+    @property
+    @abstractmethod
+    def path(self) -> Path:
+        """The path of the file the resource was loaded from"""
+
+
+class ResourceWithContext(NamedTuple):
+    """A resource and the the path to its tenant's directory
```
Comment (ernesgonzalez33) [inline tests/test_manifests.py:41]: "the" is written twice

File: tests/test_manifests.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+            for res in tenant_dir.resources:
+                ret.append(ResourceWithContext(tenant_dir.name, res))
+        return ret
+
+
+def load_tenants() -> TenantsViewer:
+    """Loads all the tenants data from disk"""
+    cur_file = Path(__file__)
+    project_dir = cur_file.parent.parent
+    cluster_dir: Path = project_dir / "auto-generated/cluster/stone-prd-rh01"
```
[the remark this answers is not in the document]
Comment (gbenhaim) [reply tests/test_manifests.py]: I remove the hardcoded cluster name.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (gbenhaim) [reply .github/workflows/test.yaml]: removed

File: tests/test_manifests.py (hunk 2 above)
Comment (gbenhaim) [reply tests/test_manifests.py]: fixed

**Pull request merged**